sql/migrate/migrate.go

```diff
@@ -499,18 +512,26 @@ func (e *Executor) Execute(ctx context.Context, m File) (err error) {
 			err = wrap(err2, err)
 		}
 	}(ctx, e.rrw, r)
-	e.log.Log(LogFile{r.Version, r.Description, r.Applied})
-	for i, stmt := range stmts {
-		if i < r.Applied {
-			// Skip if already applied.
-			continue
+	if r.Applied > 0 {
+		// If the file has been applied partially before, check if the
+		// applied history didn't change, as this is dangerous.
+		for i := 0; i < r.Applied; i++ {
```

Comment:
```suggestion
		// If the file has been applied partially before, check if the
		// applied statements were not changed.
		for i := 0; i < r.Applied; i++ {
```

No need for "dangerous". Sounds too dramatic.